lib/galaxy/jobs/__init__.py

```diff
@@ -1288,7 +1288,12 @@ def fail(self, message, exception=False, tool_stdout="", tool_stderr="", exit_co
                 dataset.mark_unhidden()
                 if dataset.ext == 'auto':
                     dataset.extension = 'data'
-                self.__update_output(job, dataset)
+                try:
+                    self.__update_output(job, dataset)
+                except:
```

Comment:
In this case I guess a bare exception probably makes sense?

Reply:
I think this should always be
```suggestion
                except Exception:
```
Otherwise we might catch `KeyboardInterrupt` or other things we shouldn't be catching.